tests/kernel/device/kernel.device.pm fails to build on cc1352r1_launchxl

**Describe the bug**
When sanitycheck is run to build for cc1352r1_launchxl, the test 'tests/kernel/device/kernel.device.pm' fails to build. This can be reproduced by building the test directly with _CONFIG_DEVICE_POWER_MANAGEMENT=y_ set.

**To Reproduce**
Steps to reproduce the behavior:
1. cd zephyr/tests/kernel/device
2. Modify prj.conf to add  _CONFIG_DEVICE_POWER_MANAGEMENT=y_
2. west build -b cc1352r1_launchxl
3. west flash
4. See error:
```
-- Build files have been written to: /home/vwan/zephyrproject-work/zephyr/tests/kernel/device/build
[95/100] Linking C executable zephyr/zephyr_prebuilt.elf
FAILED: zephyr/zephyr_prebuilt.elf 
: && ccache /home/vwan/bin/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc    zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj  -o zephyr/zephyr_prebuilt.elf  -Wl,-T zephyr/linker.cmd -Wl,-Map=/home/vwan/zephyrproject-work/zephyr/tests/kernel/device/build/zephyr/zephyr_prebuilt.map -Wl,--whole-archive app/libapp.a zephyr/libzephyr.a zephyr/arch/common/libarch__common.a zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a zephyr/lib/libc/minimal/liblib__libc__minimal.a zephyr/lib/posix/liblib__posix.a zephyr/subsys/testsuite/ztest/libsubsys__testsuite__ztest.a zephyr/drivers/gpio/libdrivers__gpio.a zephyr/drivers/serial/libdrivers__serial.a modules/ti/simplelink/source/ti/devices/cc13x2_cc26x2/lib..__modules__hal__ti__simplelink__source__ti__devices__cc13x2_cc26x2.a modules/ti/simplelink/lib..__modules__hal__ti__simplelink.a -Wl,--no-whole-archive zephyr/kernel/libkernel.a zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -L"/home/vwan/bin/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/thumb/v7e-m/nofp" -L/home/vwan/zephyrproject-work/zephyr/tests/kernel/device/build/zephyr -lgcc -Wl,--print-memory-usage -mcpu=cortex-m4 -mthumb -mabi=aapcs -Wl,--gc-sections -Wl,--build-id=none -Wl,--sort-common=descending -Wl,--sort-section=alignment -Wl,-u,_OffsetAbsSyms -Wl,-u,_ConfigAbsSyms -nostdlib -static -no-pie -Wl,-X -Wl,-N -Wl,--orphan-handling=warn && :
Memory region         Used Size  Region Size  %age Used
           FLASH:       29512 B     360360 B      8.19%
      FLASH_CCFG:          88 B         88 B    100.00%
            SRAM:        7496 B        80 KB      9.15%
        IDT_LIST:         120 B         2 KB      5.86/home/vwan/bin/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: modules/ti/simplelink/lib..__modules__hal__ti__simplelink.a(PowerCC26X2.c.obj): in function `configureXOSCHF':
/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/drivers/power/PowerCC26X2.c:1483: undefined reference to `PowerCC26X2_config'
collect2: error: ld returned 1 exit status
%
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /home/vwan/cmake/cmake-3.13.2-Linux-x86_64/bin/cmake --build /home/vwan/zephyrproject-work/zephyr/tests/kernel/device/build
```

**Expected behavior**
The test should build.

**Impact**
It prevents sanitycheck from successfully building all tests for cc1352r1_launchxl.

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - Toolchain (e.g Zephyr SDK, ...): Zephyr SDK 0.11.2
 - Commit SHA or Version used: 0fe670719941ce9a93b44d5adb39e0b262a0bf78

--- a/soc/arm/ti_simplelink/cc13x2_cc26x2/CMakeLists.txt
+++ b/soc/arm/ti_simplelink/cc13x2_cc26x2/CMakeLists.txt
@@ -6,3 +6,4 @@ zephyr_sources(soc.c)
 zephyr_sources(ccfg.c)
 
 zephyr_library_sources_ifdef(CONFIG_SYS_POWER_MANAGEMENT    power.c)
+zephyr_library_sources_ifdef(CONFIG_DEVICE_POWER_MANAGEMENT   power.c)

